qiskit_aqua/algorithms/single_sample/iqpe/iqpe.py

```diff
@@ -150,27 +151,34 @@ def init_args(self, operator, state_in, num_time_slices, num_iterations,
         self._paulis_grouping = paulis_grouping
         self._expansion_mode = expansion_mode
         self._expansion_order = expansion_order
+        self._shallow_circuit_concat = shallow_circuit_concat
         self._ret = {}
 
     def _construct_kth_evolution(self, slice_pauli_list, k, omega):
         """Construct the kth iteration Quantum Phase Estimation circuit"""
         a = QuantumRegister(1, name='a')
         c = ClassicalRegister(1, name='c')
         q = QuantumRegister(self._operator.num_qubits, name='q')
-        qc = QuantumCircuit(a, c, q)
-        qc += self._state_in.construct_circuit('circuit', q)
+        qc = self._state_in.construct_circuit('circuit', q)
         # hadamard on a[0]
+        qc.add(a)
         qc.u2(0, np.pi, a[0])
```

Comment:
do you mind changing the pi from qiskit.qasm?
```suggestion
        qc.u2(0, pi, a[0])
```

Reply:
... or vice versa

Reply:
... and done in a separate PR